service/test/BatchServerTest.cpp

```diff
@@ -426,3 +790,456 @@ TEST_F(BatchServerTest, fork_with_setup)
+         *          This function saves the backup streambuf of std::cerr, and resets it to the internal stringstream buffer.
+         */
+        void open_redirect(void)
+        {
+            if (!is_active) {
+                // Flush the remaining stderr into the terminal.
+                std::cerr << std::flush;
+                // Clear the stringstream
+                buffer.clear();
+                buffer.str(std::string());
+                // Set the streambuf of the std::cerr equivalent to the streambuf of the stringstream.
+                // Return the former streambuf of the std::cerr prior to the change.
+                cerr_backup = std::cerr.rdbuf( buffer.rdbuf() );
+                is_active = true;
+            }
+        }
+
+        /**
+         * @brief This function deactivates the redirect. It should be called after open_redirect().
+         *
+         * @details This function restores the backup streambuf associated with the stderr, and closes the stringstream redirect.
+         *
+         * @return The internal string of the stringstream, which recorded the redirected output.
+         */
+        std::string close_redirect(void)
+        {
+            if (is_active)
+            {
+                // Flush the remaining stderr into the streambuf.
+                std::cerr << std::flush;
+                // Restore the streambuf of the std::cerr to it's original state.
+                std::cerr.rdbuf(cerr_backup);
+                is_active = false;
+
+                return buffer.str();
+            } else {
+                return std::string();
+            }
+        }
+    private:
+        /// records the redirected stderr into it's own streambuf
+        std::stringstream buffer;
+        /// the backup streambuf associated with the stderr
+        std::streambuf* cerr_backup;
+        /// is true if the redirect is active
+        bool is_active;
+};
+
+/**
+ * @throws (int)2 to simulate failure of sigqueue(2) system call in POSIXSignalImp::sig_queue()
+ */
+ACTION(sig_queue_throw_2)
+{
+    throw 2;
+}
+
+/**
+ * @test Check coverage of BatchServerImp::~BatchServerImp()
+ *       Create new BatchServer objects, and have BatchServerImp::stop_batch() throw exceptions in the destructor.
+ */
+TEST_F(BatchServerTest, destructor_exceptions)
+{
+    ///
+    /// Testing catch (const Exception &ex) {}
+    ///
+
+    // Create new BatchServer object
+    std::shared_ptr<BatchServerImp> m_batch_server_exception = std::make_shared<BatchServerImp>(
+        m_client_pid,
+        m_signal_config,
+        m_control_config,
+        *m_pio_ptr,
+        m_batch_status,
+        m_posix_signal,
+        m_signal_shmem,
+        m_control_shmem,
+        m_server_pid
+    );
+
+    EXPECT_CALL(
+        *m_posix_signal.get(),
+        sig_queue(m_server_pid, SIGTERM, BatchStatus::M_MESSAGE_TERMINATE)
+    )
+    .Times(1)
+    .WillRepeatedly(sig_queue_SIGTERM());
```

Comment:
```suggestion
    EXPECT_CALL(*m_posix_signal.get(),
                sig_queue(m_server_pid, SIGTERM, BatchStatus::M_MESSAGE_TERMINATE))
        .Times(1)
        .WillRepeatedly(sig_queue_SIGTERM());
```
Please follow the style of other invocations of EXPECT_CALL.  They should all look similar to this.